eng/build.ps1

```diff
@@ -608,10 +608,6 @@ try {
 
     $global:_DotNetInstallDir = Join-Path $RepoRoot ".dotnet"
     InstallDotNetSdk $global:_DotNetInstallDir $GlobalJson.tools.dotnet
```

Comment:
This should also be unnecessary.

Reply:
Removing this breaks most of our builds

```
Listing running build processes...
Downloading vswhere
Building bootstrap compiler
Downloading RoslynTools.MSBuild 16.4.0-alpha
F:\workspace\_work\1\s\.tools\msbuild\16.4.0-alpha\tools\MSBuild\Current\Bin\msbuild.exe /nologo /bl:F:\workspace\_work\1\s\artifacts\log\Debug\Bootstrap.binlog /consoleloggerparameters:Verbosity=minimal;summary /m /nodeReuse:false /p:TreatWarningsAsErrors=true /p:Configuration=Release /p:UseRoslynAnalyzers=false /p:ContinuousIntegrationBuild=true /p:RoslynEnforceCodeStyle=false /p:UseRoslynAnalyzers=false /p:DotNetUseShippingVersions=true /p:InitialDefineConstants=BOOTSTRAP /p:PackageOutputPath=F:\workspace\_work\1\s\artifacts\Bootstrap /p:EnableNgenOptimization=false /p:PublishWindowsPdb=false /restore /t:Pack /warnaserror src\NuGet\Microsoft.Net.Compilers.Toolset\Microsoft.Net.Compilers.Toolset.Package.csproj
  It was not possible to find any installed .NET Core SDKs
  Did you mean to run .NET Core SDK commands? Install a .NET Core SDK from:
      https://aka.ms/dotnet-download
F:\workspace\_work\1\s\src\NuGet\Microsoft.Net.Compilers.Toolset\Microsoft.Net.Compilers.Toolset.Package.csproj : error : Unable to locate the .NET Core SDK. Check that it is installed and that the version specified in global.json (if any) matches the installed version.

Build FAILED.

F:\workspace\_work\1\s\src\NuGet\Microsoft.Net.Compilers.Toolset\Microsoft.Net.Compilers.Toolset.Package.csproj : error : Unable to locate the .NET Core SDK. Check that it is installed and that the version specified in global.json (if any) matches the installed version.
    0 Warning(s)
    1 Error(s)
```

Reply:
and replacing it with `InitializeDotNetCli` call?

Reply:
Tried that and it also failed. At this point want to separate this out from the PR.

Reply:
Sounds good.